admin/controllers/school.js

```diff
@@ -444,14 +444,49 @@ const getSelectPupilNotTakingCheck = async (req, res, next) => {
 
   return res.render('school/select-pupils-not-taking-check', {
     breadcrumbs: req.breadcrumbs(),
+    sortField,
+    sortDirection,
     attendanceCodes,
-    formData,
     pupilsList,
     htmlSortDirection,
     arrowSortDirection
   })
 }
 
+const savePupilNotTakingCheck = async (req, res, next) => {
+  res.locals.pageTitle = 'Save pupils not taking the check'
+  req.breadcrumbs(res.locals.pageTitle)
+
+  if (req.body.attendanceCode === undefined || req.body.pupil === undefined) {
+    return res.redirect('/school/pupils-not-taking-check/select-pupils')
+  }
+
+  const todayDate = moment(moment.now()).format()
+  let pupils = req.body.pupil
+  let pupilsProcessed
+
+  pupils.forEach(async (id) => {
+    let pupil = fetchOnePupil(id, req.user.School)
```

Comment:
We shouldn't fetch individual pupils inside a loop - we can fetch all pupils in one go which will minimise round trips.

Reply:
@pris54 this will not scale well.  Can we please do bulk operations where possible